sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -323,6 +325,8 @@ def delete_entity(
                 **kwargs
             )
         except HttpResponseError as error:
+            if error.status_code == 404 and error.reason == u"Not Found":
```

Comment:
I don't think you need the `error.reason` check - this is just a human readable version of the `status_code`.

Reply:
What we could check here is whether there's an error code in the response:
```python
error = _decode_error(error)
if error.status_code == 404  and error.error_code == 'EntityNotFound':
    # Not sure if this is an actual code value, or even if the service gives us a code at all
    return
_reraise_error(error)
```

Reply:
There's nothing in the error object, unfortunately the error object does not have an error_code parameter. I'll leave as 
```python
if error.status_code == 404:
    return
```